internal/util/util.go

```diff
@@ -93,11 +94,11 @@ type TypePair struct {
 // SamplingStrategy represents the config to use for sampling golden response
 // Json.
 type SamplingStrategy struct {
-	// Sampling ratio for list and map.
-	Ratio float32
+	// Maximum number of samples. -1 means sample all.
```

Comment:
can you describe here how the samples are generated based on this value? i think there's still value in picking up new data (e.g. for covid, if we sampled every n). with max samples, we could quickly exhaust it and not see new data, if all samples are kept from the start of the list / map.

Reply:
The sampling is uniform and start from the end. For example, sample from [1,2,3,4,5,6,7,8,9,10] with MaxSample = 4, would result in [1,4,7,10]

Reply:
thanks, could you add to the comment?

Reply:
Done